lib/script_base.rb

```diff
@@ -0,0 +1,98 @@
+require 'optparse'
+
+class ScriptBase
```

Comment:
I'm open to receiving feedback on the chosen class name. Feel free to share your thoughts or suggestions on alternative class names.